nim/tests/test_routes.py

```diff
@@ -0,0 +1,103 @@
+from nim.app import app
+import pytest
+import json
+
+
+@pytest.fixture
+def test_app():
+    return app.test_client()
+
+
+def test_index(test_app):
+    response = test_app.get('/', content_type='html/text')
+    assert response.status_code == 200
+
+
+def test_new_success(test_app):
```

Comment:
Most of the functions below should be factorized, it should make it much easier to test all those edge cases.

Reply:
What exactly do you mean by 'factorized'?

Reply:
For example, take `test_new_max_limit_fail`, `test_new_piles_limit_fail`, `test_new_all_limit_fail`, `test_new_pile_zero_fail`, `test_new_all_zero_fail`, `test_new_non_int_pile_fail` and `test_new_non_int_all_fail`.
Their code are almost exactly the same, so you could factorize them.

If you're not familiar with the term, is means simplifying your code by creating bits of code that you can re-use. For example, let's say you have the following:
```
def f1(n):
    print(f"Computing {n} + 1")
    res = n + 1
    print(f"{n} + 1 = {res}")
    return res

def f2(n):
    print(f"Returning {n} + 2")
    res = n + 2
    print(f"{n} + 2 = {res}")
    return res

def f3(n):
    print(f"Returning {n} + 3")
    res = n + 3
    print(f"{n} + 3 = {res}")
    return res

def f4(n):
    print(f"Returning {n} + 4")
    res = n + 4
    print(f"{n} + 4 = {res}")
    return res

def f5(n):
    print(f"Returning {n} + 5")
    res = n + 5
    print(f"{n} + 5 = {res}")
    return res
```

That's fine and all, but look at all that code being copied and pasted over and over! There has to be a way to clean that up. Let's factorize our code!

```
def f(n, inc):
    print(f"Returning {n} + {inc}")
    res = n + inc
    print(f"{n} + {inc} = {res}")
    return res

def f1(n):
    return f(n, 1)

def f2(n):
    return f(n, 2)

def f3(n):
    return f(n, 3)

def f4(n):
    return f(n, 4)

def f5(n):
    return f(n, 5)
```

Much cleaner! Now if you need to modify what is being printed by example, you only need to do it in one single place instead of 5 before.

You can apply the same logic to your functions here.

Reply:
Cool, that's what I thought you meant but wanted to make sure 🙂 
Nope, I was not familiar with the term